firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -50,6 +50,15 @@
   // Remove this constant and increment VERSION to enable indexing support
   static final int INDEXING_SUPPORT_VERSION = VERSION + 1;
 
+  /**
+   * The batch size for the sequence number migration in `ensureSequenceNumbers()`.
+   *
+   * <p>This addresses https://github.com/firebase/firebase-android-sdk/issues/370, where a customer
+   * reported that schema migrations failed for clients with thousands of documents. The number has
+   * been chosen arbitrarily.
+   */
+  private static final int SEQUENCE_NUMBER_BATCH_SIZE = 1000;
```

Comment:
According to https://cloud.google.com/firestore/quotas, the maximum length of a path (document name) is 6 KiB.  Can you try your repro with 6K paths and the 1000 batch size and make sure it still doesn't hit the error?

Also, if we were to change this number in the future, how would one verify it is safe?  Our automated tests aren't enough, right?  So maybe throw your repro code in a gist and link it from here so somebody in the future could repro?  So then this comment would change to "This number was chosen somewhat arbitrarily but has been verified to avoid the reported issue, using the repro code at [your gist]." or whatever.

Reply:
My verification steps use FriendlyEats and code that inserts documents in batches. There is no specific magic and most of the work revolves around restarting the client with different version of the SDK.  I don't feel it's worth linking to a code snippet. 
From my experience with SQLite, it also seems impossible to come up with a number that "just works" and increasing the document name to 6 KB ended up throwing a different error (" java.lang.IllegalStateException: Couldn't read row 114, col 0 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.") with this batch size. Without further ado, I reduced the batch size back to 100, which works even when I use 6KB document names.